backend/s3/s3.go

```diff
@@ -3837,7 +3839,8 @@ func (o *Object) downloadFromURL(ctx context.Context, bucketPath string, options
 	for key, value := range resp.Header {
 		if strings.HasPrefix(key, "x-amz-meta") {
 			metaKey := strings.TrimPrefix(key, "x-amz-meta-")
-			metaData[strings.Title(metaKey)] = &value[0]
+			metaKey = cases.Title(language.Und, cases.NoLower).String(metaKey)
```

Comment:
Gah, this must be wrong! Your change is fine, but the code is wrong before. I suggest you leave this one, and I will fix it properly in my metadata branch!